external-crates/move/crates/move-compiler/tests/move_2024/expansion/macro_identifier_invalid_type_parameter.move

```diff
@@ -0,0 +1,4 @@
+module a::m {
+    // invalid type parameter
+    macro fun foo<$_, $__, $_t>() {}
```

Comment:
Can you add tests for `__` (without a dollar) usage too?

Reply:
It is covered in a separate test

Reply:
Nope. I lied. We currently allow all sorts of nonsense, e.g. `fun foo<__, ___T>` 🙃

Reply:
Ban them in Move 2024 and add them to the migration tooling?

Reply:
> Ban them in Move 2024 and add them to the migration tooling?

Too late. And I'm not sure. They are gross for sure, and maybe we should just warn/lint against them instead. We can always ban it in some future version if needed